lib/utils/getFormatter.mjs

```diff
@@ -0,0 +1,64 @@
+import { EXIT_CODE_INVALID_CONFIG } from '../constants.mjs';
+import dynamicImport from './dynamicImport.mjs';
+import { existsSync } from 'node:fs';
+import formatters from '../formatters/index.mjs';
+import getConfigForFile from '../getConfigForFile.mjs';
+import getFormatterOptionsText from './getFormatterOptionsText.mjs';
+import { resolve } from 'node:path';
+
+/** @import {Formatter, FormatterType, InternalApi} from 'stylelint' */
+
+/**
+ * @param {InternalApi} stylelint
+ * @param {Object} [params]
+ * @param {FormatterType | Formatter | undefined} [params.formatterType]
+ * @returns {Promise<Formatter>}
+ */
+export default async function getFormatter(stylelint, params = {}) {
+	const { formatterType } = params;
+	const cwd = stylelint._options.cwd;
+	const configPath = stylelint._options.configFile || cwd;
+	let formatter = formatterType;
```

Comment:
[suggestion] I think we can simplify this function logic without `params`.

First, it tries getting a formatter from linter options (i.e., `standalone()` arguments):

```suggestion
	let formatter = stylelint._options.formatter;
```